Sources/Extensions/Alamofire/AlamofireRequest+Extensions.swift

```diff
@@ -90,11 +92,60 @@ public extension Reactive where Base: DataRequest {
 
                     return Observable.zip(createFromList) { $0 }
                         .map { (resp, $0) }
+                }
+    }
+
+    internal func responseJSONOnQueue(_ queue: DispatchQueue) -> Observable<ServerResponse> {
+        let responseSerializer = DataRequest.jsonResponseSerializer(options: .allowFragments)
+
+        return responseResult(queue: queue, responseSerializer: responseSerializer)
+            .catchError {
+                switch $0 {
+                case let urlError as URLError:
+                    switch urlError.code {
+                    case .notConnectedToInternet, .timedOut:
+                        throw RequestError.noConnection
+                    default:
+                        throw RequestError.network(error: urlError)
+                    }
+                case let afError as AFError:
+                    switch afError {
+                    case .responseSerializationFailed, .responseValidationFailed:
+                        throw RequestError.invalidResponse(error: afError)
+                    default:
+                        throw RequestError.network(error: afError)
+                    }
+                default:
+                    throw RequestError.network(error: $0)
+                }
             }
     }
 
-    internal func responseJSONOnQueue(_ queue: DispatchQueue) -> Observable<(HTTPURLResponse, Any)> {
-        return responseResult(queue: queue, responseSerializer: DataRequest.jsonResponseSerializer(options: .allowFragments))
+}
+
+private extension ObservableType where E == ServerResponse {
+
+    func tryMapResult<R>(_ transform: @escaping (E) throws -> R) -> Observable<R> {
+        return map {
+            do {
+                return try transform($0)
+            } catch {
+                throw RequestError.mapping(error: error, response: $0.1)
```

Comment:
Unnamed tuple :)

Reply:
It comes from RxAlamofire `responseResult(queue:responseSerializer:)`.
The only thing that I can do it:
```swift
.map { ServerResponse(response: $0.0, result: $0.1) }
```